src/Libraries/CoreNodeModelsWpf/SliderViewModel.cs

```diff
@@ -63,10 +63,18 @@ public T Value
                 if (value.CompareTo(model.Min) == -1)
                     model.Min = value;
 
-                if(value is IFormattable formattableval)
+                var stepValueString = model.Step.ToString();
+                var decimalPoints = 0;
+                if (stepValueString.Contains('.'))
+                {
+                    decimalPoints = stepValueString.Substring(stepValueString.IndexOf('.') + 1).Length;
+                }
+
+                if (value is IFormattable formattableval)
                 {
                     var invariantString  = formattableval.ToString(null,CultureInfo.InvariantCulture);
-                    model.UpdateValue(new Dynamo.Graph.UpdateValueParams(nameof(Value), invariantString));
+                    var sliderValue = Math.Round(decimal.Parse(invariantString), decimalPoints);
+                    model.UpdateValue(new Dynamo.Graph.UpdateValueParams(nameof(Value), sliderValue.ToString()));
                 }
                 else
                 {
```

Comment:
Does anyone know in which case the else block here gets triggered? I am thinking if we should round the value there as well?

Reply:
I can't think of any. I believe `int`, `float`, `double`, `long`, are all `IFormattable`.